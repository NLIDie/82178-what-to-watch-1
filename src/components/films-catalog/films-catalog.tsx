import * as React from "react";

// Components
import {FilmsGenresNavList} from "_Components/films-genres-nav-list/films-genres-nav-list";
import {FilmCardList} from "_Components/film-card-list/film-card-list";
import {ButtonShowMore} from "_Components/button-show-more/button-show-more";

// Types
import {TProps} from "./films-catalog.types";

export function FilmsCatalog(props: TProps): JSX.Element {
  const {genres, onChangeGenre, films, onClickShowMore, canShowMore} = props;

  return (
    <section className="catalog">
      <h2 className="catalog__title visually-hidden">Catalog</h2>

      <FilmsGenresNavList genres={genres} onChange={onChangeGenre} />

      <FilmCardList films={films} />

      {canShowMore && <ButtonShowMore onClick={onClickShowMore} />}
    </section>
  );
}
