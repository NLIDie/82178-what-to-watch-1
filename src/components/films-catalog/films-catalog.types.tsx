// Types
import {TFilm} from "_Types/film";

export type TProps = {
  readonly genres: TFilm['genre'][];
  readonly films: TFilm[];
  readonly canShowMore: boolean;
  readonly onChangeGenre: (genre: TFilm['genre']) => void;
  readonly onClickShowMore: (
    event: React.SyntheticEvent<HTMLButtonElement>
  ) => void;
};
