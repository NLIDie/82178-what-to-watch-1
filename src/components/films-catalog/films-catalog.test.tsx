import * as React from "react";
import {create} from "react-test-renderer";

// Components
import {FilmsCatalog} from "./films-catalog";

jest.mock(`react-router-dom`, () => ({
  BrowserRouter: ({children}) => <div>{children}</div>,
  Route: ({children}) => <div>{children}</div>,
  Link: ({children}) => <a>{children}</a>
}));

describe(`Components/FilmCardTabReviewsContent`, () => {
  it(`Correct renderer the component`, () => {
    const tree = create(
        <FilmsCatalog
          films={[
            {
              backgroundColor: `#D8E3E5`,
              backgroundImage:
              `https://es31-server.appspot.com/wtw/static/film/background/Moonrise_Kingdom.jpg`,
              description:
              `A pair of young lovers flee their New England town, which causes a local search party to fan out to find them.`,
              director: `Wes Anderson`,
              genre: `Adventure`,
              id: 1,
              isFavorite: false,
              name: `Moonrise Kingdom`,
              posterImage:
              `https://es31-server.appspot.com/wtw/static/film/poster/Moonrise_Kingdom.jpg`,
              previewImage:
              `https://es31-server.appspot.com/wtw/static/film/preview/moonrise-kingdom.jpg`,
              previewVideoLink:
              `https://upload.wikimedia.org/wikipedia/commons/transcoded/b/b3/Big_Buck_Bunny_Trailer_400p.ogv/Big_Buck_Bunny_Trailer_400p.ogv.360p.webm`,
              rating: 3.9,
              released: 2012,
              runTime: 94,
              scoresCount: 291183,
              starring: [`Jared Gilman`, `Kara Hayward`, `Bruce Willis`],
              videoLink:
              `http://peach.themazzone.com/durian/movies/sintel-1024-surround.mp4`
            }
          ]}
          genres={[`genre1`, `genre2`, `genre3`]}
          onChangeGenre={jest.fn()}
          onClickShowMore={jest.fn()}
          canShowMore={false}
        />
    );

    expect(tree).toMatchSnapshot();
  });
});
