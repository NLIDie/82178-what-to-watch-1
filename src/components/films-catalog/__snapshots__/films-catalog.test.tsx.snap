// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Components/FilmCardTabReviewsContent Correct renderer the component 1`] = `
<section
  className="catalog"
>
  <h2
    className="catalog__title visually-hidden"
  >
    Catalog
  </h2>
  <ul
    className="catalog__genres-list"
  >
    <li
      className="catalog__genres-item catalog__genres-item--active"
      onClick={[Function]}
    >
      <a
        className="catalog__genres-link"
        href="#"
      >
        genre1
      </a>
    </li>
    <li
      className="catalog__genres-item "
      onClick={[Function]}
    >
      <a
        className="catalog__genres-link"
        href="#"
      >
        genre2
      </a>
    </li>
    <li
      className="catalog__genres-item "
      onClick={[Function]}
    >
      <a
        className="catalog__genres-link"
        href="#"
      >
        genre3
      </a>
    </li>
  </ul>
  <div
    className="catalog__movies-list"
  >
    <article
      className="small-movie-card catalog__movies-card"
    >
      <a>
        <video
          height="176"
          muted={true}
          onMouseOut={[Function]}
          onMouseOver={[Function]}
          poster="https://es31-server.appspot.com/wtw/static/film/preview/moonrise-kingdom.jpg"
          src="https://upload.wikimedia.org/wikipedia/commons/transcoded/b/b3/Big_Buck_Bunny_Trailer_400p.ogv/Big_Buck_Bunny_Trailer_400p.ogv.360p.webm"
          width="280"
        />
      </a>
      <h3
        className="small-movie-card__title"
      >
        <a>
          Moonrise Kingdom
        </a>
      </h3>
    </article>
  </div>
</section>
`;
